h3/src/quic.rs

```diff
@@ -79,7 +82,7 @@ pub trait RecvStream {
     /// The type of `Buf` for data received on this stream.
     type Buf: Buf;
     /// The error type that can occur when receiving data.
-    type Error; // bounds?
+    type Error: Into<io::Error>;
```

Comment:
@seanmonstar What do you think of this bound ? Quinn returns io::Error, and I think this fits the use case. 

It makes calling `.into()` twice in `h3::frame` though. So it makes me feel there is a better solution?

Reply:
Yea I wasn't sure, thus the question in comments. In hyper, I've required `Into<Box<dyn std::error::Error + Send + Sync>>`, which allows me to wrap the error with some context in hyper, and still provide access to the original in `Error::source()`.

I'd probably lean towards that, but wasn't sure if there was something better.

Reply:
So we would require `dyn std::error::Error` if we want to add some context at the `h3` layer ? `Send + Sync` is also needed right ?

Reply:
Yes, If any, I'd add `type Error: Into<Box<dyn std::error::Error + Send + Sync>>`.

Reply:
Ok thanks. I'm not very knowledgeable when it comes to choose API bounds upfront for the moment.

Reply:
Wouldn't `std::io::Error` have a large enough field of possibilities, instead of `std::error::Error` ?

Reply:
Typically `io::Error` is annoying, since the "cause" kind of *becomes* the `io::Error`, but only partly.

Doing as I said tends to be more flexible and clearer to users. But if it's easier, we can just merge with this and I can follow up with an error design PR.

Reply:
I'll try to get this done in the current PR.

Reply:
Done.